templates/repo/issue/search.tmpl

```diff
@@ -7,7 +7,7 @@
 		<input type="hidden" name="project" value="{{$.ProjectID}}">
 		<input type="hidden" name="assignee" value="{{$.AssigneeID}}">
 		<input type="hidden" name="poster" value="{{$.PosterID}}">
-		<input name="q" value="{{.Keyword}}" placeholder="{{.locale.Tr "explore.search"}}..." maxlength="255">
+		{{template "shared/searchinput" dict "." . "Value" .Keyword}}
```

Comment:
It could use ` "locale" .locale ` instead of passing the big map  for every call.

It's quite fragile to pass the who unclear current object into the sub-template.

For example, what if there is a "AutoFocus" or "Disabled" in `.`? It breaks the input state.

Reply:
Hmm yes, we can do that as the template is quite small and does not need any other vars.

Reply:
> For example, what if there is a "AutoFocus" or "Disabled" in .? It breaks the input state.

I guess it would not break because latter `dict` vars override earlier ones, no?

Reply:
```
.Disabled=true
{{template "shared/searchinput" dict "." . "Value" .Keyword}}
```

Then, do you want to render a disabled input?

Reply:
That would be a bug, yes. I guess in other cases where we pass `.`, we should ensure the names are unique enough.

locale change done in https://github.com/go-gitea/gitea/pull/25338/commits/531c6f017518dd7d86704c065c8662bb913287be.

Reply:
![image](https://github.com/go-gitea/gitea/assets/2114189/a873e173-143a-4bf0-b674-7ffae8390eb0)

Reply:
Right. i guess for small templates, I definitely wouldn't use `.`. But for big ones with lot of vars, I think it's cleaner.